src/Symfony/Component/Serializer/Tests/Encoder/XmlEncoderTest.php

```diff
@@ -262,6 +262,24 @@ public function testDecodeNegativeFloatAttribute()
         $this->assertSame(array('@index' => -12.11, '#' => 'Name'), $this->encoder->decode($source, 'xml'));
     }
 
+    public function testNoTypeCastAttribute()
+    {
+        $source = <<<XML
+<?xml version="1.0"?>
+<document a="018" b="-12.11">
+    <node a="018" b="-12.11"/>
+</document>
+XML;
+
+        $data = $this->encoder->decode($source, 'xml', array('xml_type_cast_attributes' => false));
+        $this->assertTrue(
+            $data['@a'] === '018' &&
+            $data['@b'] === '-12.11' &&
+            $data['@a'] === $data['node']['@a'] &&
+            $data['@b'] === $data['node']['@b']
```

Comment:
I think these needs to be separate asserts (you can then probably use `assertSame`)